File: backend/src/main/java/com/shuike/manager/modules/teachingplan/entity/TeachingPlan.java
package com.shuike.manager.modules.teachingplan.entity;

import com.baomidou.mybatisplus.annotation.*;
import lombok.Data;
import java.time.LocalDateTime;

@Data
@TableName("teaching_plans")
public class TeachingPlan {
    @TableId(type = IdType.AUTO)
    private Long id;
    private Long teacherId;
    private Long courseId;
    private Long semesterId;
    private String classInfo;
    private String textbookInfo;
    private String summary;
    private String status;
    private LocalDateTime submitTime;
    private LocalDateTime collegeReviewTime;
    private LocalDateTime officeReviewTime;
    @TableField(fill = FieldFill.INSERT)
    private LocalDateTime createdAt;
    @TableField(fill = FieldFill.INSERT_UPDATE)
    private LocalDateTime updatedAt;
}

File: frontend/src/types/teachingPlan.ts
export interface TeachingPlan {
  id?: number
  teacherId?: number
  courseId?: number
  semesterId?: number
  classInfo?: string
  textbookInfo?: string
  summary?: string
  status?: string
  submitTime?: string
  collegeReviewTime?: string
  officeReviewTime?: string
  createdAt?: string
  updatedAt?: string
}

export interface TeachingPlanFile {
  id?: number
  planId?: number
  fileName: string
  fileUrl: string
  fileType: string
  fileSize: number
}
